packages/core/src/domain/conversion/__tests__/cleanup.test.ts

```diff
@@ -199,6 +199,18 @@ describe("replaceNullFlavor", () => {
     expect(replaceNullFlavor(input)).toBe(expected);
   });
 
+  test("should keep the self-closing tag", () => {
+    const input = '<id nullFlavor = "NI"/>';
+    const expected = '<id extension="1" root="1"/>';
+    expect(replaceNullFlavor(input)).toBe(expected);
+  });
+
+  test("should handle nullFlavor id that still contains an extension", () => {
+    const input = '<id nullFlavor="NI" extension"12345">';
+    const expected = '<id extension="1" root="1">';
+    expect(replaceNullFlavor(input)).toBe(expected);
+  });
```

Comment:
Do these represent well the situations you found IRL? Though there was something else, but it might be covered in one of those.